File: app/src/main/java/com/sdex/commons/ads/AppPreferences.java
package com.sdex.commons.ads;

import android.content.Context;
import android.content.SharedPreferences;

public class AppPreferences {

  private static final String PREFERENCES_NAME = "ads_preferences";
  private static final String KEY_PRO = "pro";
  private static final String KEY_HISTORY_WARNING_SHOWN = "history_warning_shown";
  private static final String KEY_NOT_EXPORTED_DIALOG_SHOWN = "not_exported_dialog_shown";

  private final SharedPreferences preferences;

  public AppPreferences(Context context) {
    preferences = context.getSharedPreferences(PREFERENCES_NAME, Context.MODE_PRIVATE);
  }

  public void setProVersion(boolean isPro) {
    preferences.edit()
      .putBoolean(KEY_PRO, isPro)
      .apply();
  }

  public boolean isProVersion() {
    return preferences.getBoolean(KEY_PRO, false);
  }

  public SharedPreferences getPreferences() {
    return preferences;
  }

  public boolean isHistoryWarningShown() {
    return preferences.getBoolean(KEY_HISTORY_WARNING_SHOWN, false);
  }

  public void setHistoryWarningShown(boolean historyWarningShown) {
    preferences.edit()
      .putBoolean(KEY_HISTORY_WARNING_SHOWN, historyWarningShown)
      .apply();
  }

  public boolean isNotExportedDialogShown() {
    return preferences.getBoolean(KEY_NOT_EXPORTED_DIALOG_SHOWN, false);
  }

  public void setNotExportedDialogShown(boolean notExportedDialogShown) {
    preferences.edit()
      .putBoolean(KEY_NOT_EXPORTED_DIALOG_SHOWN, notExportedDialogShown)
      .apply();
  }
}

File: app/src/main/java/com/sdex/commons/ads/AppPreferences.kt
package com.sdex.commons.ads

import android.content.Context
import android.content.SharedPreferences

class AppPreferences(context: Context) {

  val preferences: SharedPreferences

  var isProVersion: Boolean
    get() = preferences.getBoolean(KEY_PRO, false)
    set(isPro) = preferences.edit()
      .putBoolean(KEY_PRO, isPro)
      .apply()

  var isHistoryWarningShown: Boolean
    get() = preferences.getBoolean(KEY_HISTORY_WARNING_SHOWN, false)
    set(historyWarningShown) = preferences.edit()
      .putBoolean(KEY_HISTORY_WARNING_SHOWN, historyWarningShown)
      .apply()

  var isNotExportedDialogShown: Boolean
    get() = preferences.getBoolean(KEY_NOT_EXPORTED_DIALOG_SHOWN, false)
    set(notExportedDialogShown) = preferences.edit()
      .putBoolean(KEY_NOT_EXPORTED_DIALOG_SHOWN, notExportedDialogShown)
      .apply()

  init {
    preferences = context.getSharedPreferences(PREFERENCES_NAME, Context.MODE_PRIVATE)
  }

  companion object {

    private val PREFERENCES_NAME = "ads_preferences"
    private val KEY_PRO = "pro"
    private val KEY_HISTORY_WARNING_SHOWN = "history_warning_shown"
    private val KEY_NOT_EXPORTED_DIALOG_SHOWN = "not_exported_dialog_shown"
  }
}
